api/api/services/documentation_service.py

```diff
@@ -56,8 +68,73 @@ def get_all_doc_sections(self) -> list[DocumentationSection]:
                     )
         return doc_sections
 
-    def get_documentation_by_path(self, pathes: list[str]) -> list[DocumentationSection]:
-        all_doc_sections: list[DocumentationSection] = self.get_all_doc_sections()
+    async def _get_all_doc_sections_remote(self) -> list[DocumentationSection]:
+        """Fetch all documentation sections from remote fumadocs instance"""
+        doc_sections: list[DocumentationSection] = []
+
+        async with httpx.AsyncClient() as http_client:
+            try:
+                _logger.info("Fetching documentation sections from remote fumadocs instance")
+                url = f"{WORKFLOWAI_DOCS_URL}/api/ai/index"
+                response = await http_client.get(url)
+                response.raise_for_status()
+
+                tasks = [self._fetch_page_content(page_raw["url"].lstrip("/")) for page_raw in response.json()["pages"]]
+                page_contents = await asyncio.gather(*tasks)
+
+                for page_raw, page_content in zip(response.json()["pages"], page_contents):
+                    page_title = page_raw["url"].lstrip("/")
+                    doc_sections.append(
+                        DocumentationSection(title=page_title, content=page_content),
+                    )
+
+            except Exception as e:
+                _logger.exception(
```

Comment:
ok to just fail here as well ? Otherwise the tool will get that sections are empty but without an error